addons/base_import_module/models/ir_http.py

```diff
@@ -0,0 +1,57 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import io
+
+from odoo import api, models
+from odoo.tools import ormcache
+from odoo.tools.translate import CodeTranslations, get_base_langs, JAVASCRIPT_TRANSLATION_COMMENT
+
+
+class IrHttp(models.AbstractModel):
+    _inherit = 'ir.http'
+
+    @api.model
+    @ormcache('module', 'lang')
+    def _get_trasnlations_for_webclient_imported_module(self, module, lang):
+        if not lang:
+            lang = self.env.context.get("lang") or 'en_US'
+        IrAttachment = self.env['ir.attachment']
+
+        def filter_func(row):
+            return row.get('value') and JAVASCRIPT_TRANSLATION_COMMENT in row['comments']
+
+        translations = {}
+
+        for lang_ in get_base_langs(lang):
+            attachment = IrAttachment.sudo().search([
+                    ('name', '=', f"{module}_{lang_}.po"),
+                    ('url', '=', f"/{module}/i18n/{lang_}.po"),
+                    ('type', '=', 'binary'),
+                ], limit=1)
+            if attachment.raw:
+                # open lang_attachment.data as a file object
+                with io.BytesIO(attachment.raw) as fileobj:
+                    fileobj.name = attachment.name
+                    translations.update(CodeTranslations._read_code_translations_file(fileobj, filter_func))
+
+        return {
+            'messages': tuple({
+                'id': src,
+                'string': value,
+            } for src, value in translations.items())
+        }
+
+    @api.model
+    def get_translations_for_webclient(self, modules, lang):
+        all_imported_modules = self.env['ir.module.module']._get_imported_module_names()
+        if not modules:
+            non_imported_modules = []
+            imported_modules = all_imported_modules
+        else:
+            non_imported_modules = [m for m in modules if m not in all_imported_modules] or ['__dummy__']
+            imported_modules = [m for m in modules if m in all_imported_modules]
+
+        translations_per_module, lang_params = super().get_translations_for_webclient(non_imported_modules, lang)
+        for module in imported_modules:
+            translations_per_module[module] = self._get_trasnlations_for_webclient_imported_module(module, lang)
```

Comment:
Translations for each module will be merged to a huge object in js
https://github.com/odoo-dev/odoo/blob/9ad626e9782dec68f722a010ea080f7cabc468d0/addons/web/static/src/core/l10n/localization_service.js#L54

The order of the keys(module names) is not strictly promised by json, but I think it is kept in practice.
As a result, this implementation brings a side effect, which allows users to import a customized module to override any webclient translation which belongs to another non-imported module.